pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,353 @@
+
+        Raises:
+            KeyError: If no object with name exists in the object store.
+        """
+        pass
+
+    @abstractmethod
+    def delete(self, name):
+        """
+        Delete the object identified by name from the object store.
+
+        Parameters:
+
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store.
+
+        Raises:
+            KeyError: If there is no object with name in this object store
+        """
+        pass
+
+    @abstractmethod
+    def iter_names(self):
+        """
+        Return iterator to the names of the objects in the object store.
+
+        Returns:
+            List of the names of CIM objects in the object store
+        """
+        pass
+
+    @abstractmethod
+    def iter_values(self, copy=True):
+        """
+        Return iterator to the cim objects in the object store. This allows
+        iteration through all the objects in this
+        object store. Objects may be accessed using iterator methods.
+
+        Parameters:
+          copy (:class:`py:bool`):
+            Copy the objects before returning them.  This is the default
+            behavior and also the mode that should be used unless the
+            user is certain the object will NOT be modified after it is
+            returned.
+
+        Returns:
+            Python iterator for the object values in the object store.
+            If copy == True, each object is copied before it is returned.
+        """
+        pass
+
+    @abstractmethod
+    def len(self):
+        """
+        Get count of objects in this object store.
+
+        Returns:
+            Integer that is count of objects in this objectstore
+
+        """
+        pass
+
+
+@six.add_metaclass(ABCMeta)
+class BaseRepositoryAPI(object):
+    """
+    An abstract base class defining the required  APIs to provide access to a
+    data store for a CIM repository.  The API provides functions to:
+
+    1. Manage CIM namespaces in the data repository including creation, deletion
+       and getting a list of the available namespaces.
+    2. Accessing the object store within the repository for the objects of the
+       following CIM types: (CIM classes, CIM instances, and CIM qualifier
+       decelarations) so that methods of the ObjStoreAPI can be used to access
+       objects by namespace so that CIM objects can be manupulated in the
+       repository by namespace.
+    """
+
+    @compatibleabstractproperty
+    def namespaces(self):
+        """
+        Readonly property that returns a list with the names of the
+        namespaces defined for this repository
+
+        Returns:
+            list of :term:`string` items containing the namespace names
+        """
+        pass
+
+    @abstractmethod
+    def validate_namespace(self, namespace):
+        """
+        Validate if the namespace parameter exists in the repository. If the
```

Comment:
It validates whether a namespace exists, not the namespace parameter.

Reply:
Fixed DONE